app/src/test/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModelTest.kt

```diff
@@ -414,6 +416,36 @@ class GroupConversationDetailsViewModelTest {
         assertEquals(expected, viewModel.conversationSheetContent)
     }
 
+    @Test
+    fun `given receipt mode value enabled, when updating receipt mode, then value is propagated to screen state`() = runTest {
+        // given
+        val (_, viewModel) = GroupConversationDetailsViewModelArrangement()
+            .withUpdateConversationReceiptModeReturningSuccess()
+            .arrange()
+        val receiptModeEnabled = true
+
+        // when
+        viewModel.onReadReceiptUpdate(enableReadReceipt = receiptModeEnabled)
+
+        // then
+        assertEquals(receiptModeEnabled, viewModel.groupOptionsState.value.isReadReceiptAllowed)
+    }
+
+    @Test
```

Comment:
Maybe we can `verify` that the mocked usecase is called, so we know later if something is not called, we can catch it 🦋